Remove work around for DisplayOperatorMinHeight

Ahmad-S792 opened the pull request:
#### 84ef2457635ee078de49ee79e48cacfeb7604b97
<pre>
Remove work around for DisplayOperatorMinHeight
<a href="https://bugs.webkit.org/show_bug.cgi?id=297646">https://bugs.webkit.org/show_bug.cgi?id=297646</a>
<a href="https://rdar.apple.com/159260284">rdar://159260284</a>

Reviewed by Frédéric Wang.

This patch aligns WebKit with Gecko / Firefox and Web Specification [1]:

This patch removes workaround when DisplayOperatorMinHeight is too small
(e.g., in case of Cambria Math font on Windows), for non-Apple ports, we
have updated harfbuzz with fix. So this workaround was not necessary anymore.

It progresses web platform test, which shows that we are not compensating
for the constant being too small.

[1] <a href="https://w3c.github.io/mathml-core/#layout-of-operators">https://w3c.github.io/mathml-core/#layout-of-operators</a>

* Source/WebCore/rendering/mathml/MathOperator.cpp:
(WebCore::MathOperator::calculateDisplayStyleLargeOperator):
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/display-operator-min-height-no-workaround-expected.txt: Progress

Canonical link: <a href="https://commits.webkit.org/299773@main">https://commits.webkit.org/299773@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/347d21eac066e716706a459944301025cbf8b45f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120121 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39815 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30466 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126469 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72194 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e06505dc-4e09-47ab-bbc7-25dc16f02b75) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121997 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40511 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48392 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/91227 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/60536 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ed3b03a6-d6d6-46d9-b4c7-c045abde466e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123073 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32362 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107705 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71778 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bd969d5a-20f8-4bd4-bcb8-d9963be85347) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/31394 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25811 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70096 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101841 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25998 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129374 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47041 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/35685 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/99844 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47407 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103893 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99687 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25310 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45145 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23167 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/43672 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46903 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52609 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46371 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49718 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48055 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

fred-wang commented on Source/WebCore/rendering/mathml/MathOperator.cpp:
```diff
@@ -242,8 +242,7 @@ void MathOperator::calculateDisplayStyleLargeOperator(const RenderStyle& style)
 
-    // The value of displayOperatorMinHeight is sometimes too small, so we ensure that it is at least \sqrt{2} times the size of the base glyph.
-    float displayOperatorMinHeight = std::max(heightForGlyph(baseGlyph) * std::numbers::sqrt2_v<float>, baseGlyph.font->mathData()->getMathConstant(*baseGlyph.font, OpenTypeMathData::DisplayOperatorMinHeight));
+    float displayOperatorMinHeight = baseGlyph.font->mathData()->getMathConstant(*baseGlyph.font, OpenTypeMathData::DisplayOperatorMinHeight);
```
Is the `#include <numbers>` still needed in this file?

Ahmad-S792 replied:
Let me check locally, if not, I will remove it. :-)

fred-wang reviewed (approved):
LGTM, thank you!

Ahmad-S792 committed:
Remove work around for DisplayOperatorMinHeight
https://bugs.webkit.org/show_bug.cgi?id=297646
rdar://159260284

Reviewed by Frédéric Wang.

This patch aligns WebKit with Gecko / Firefox and Web Specification [1]:

This patch removes workaround when DisplayOperatorMinHeight is too small
(e.g., in case of Cambria Math font on Windows), for non-Apple ports, we
have updated harfbuzz with fix. So this workaround was not necessary anymore.

It progresses web platform test, which shows that we are not compensating
for the constant being too small.

[1] https://w3c.github.io/mathml-core/#layout-of-operators

* Source/WebCore/rendering/mathml/MathOperator.cpp:
(WebCore::MathOperator::calculateDisplayStyleLargeOperator):
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/display-operator-min-height-no-workaround-expected.txt: Progress

Canonical link: https://commits.webkit.org/299773@main
